packages/browser-destinations/src/destinations/heap/trackEvent/index.ts

```diff
@@ -56,20 +56,59 @@ const action: BrowserActionDefinition<Settings, HeapApi, Payload> = {
     }
   },
   perform: (heap, event) => {
-    const eventProperties = Object.assign({}, event.payload.properties)
-    eventProperties.segment_library = HEAP_SEGMENT_BROWSER_LIBRARY_NAME
+    // Add user properties
     if (event.payload.anonymousId || isDefined(event.payload?.traits)) {
       const traits = flat(event.payload?.traits)
       heap.addUserProperties({
         ...(isDefined(event.payload.anonymousId) && { anonymous_id: event.payload.anonymousId }),
         ...(isDefined(traits) && traits)
       })
     }
+
+    // Identify user
     if (event.payload?.identity && isDefined(event.payload?.identity)) {
       heap.identify(event.payload.identity)
     }
-    heap.track(event.payload.name, eventProperties)
+
+    // Track Events
+    const eventProperties = Object.assign({}, event.payload.properties)
+    const eventName = event.payload.name
+    const arrayProperties: { [key: string]: Properties[] } = {}
+
+    for (const [key, value] of Object.entries(eventProperties)) {
```

Comment:
This should be merged with the loop on line 87 for performance.

Reply:
@murphpdx What do you mean, how would we merge them? I thought we'd need to loop through all properties, extract arrays, and then loop through those arrays to track the "items"